debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/processor/memory/MemoryLogMinerEventProcessor.java

```diff
@@ -215,6 +215,14 @@ protected void handleCommit(LogMinerEventRow row) throws InterruptedException {
             return;
         }
 
+        if (transaction.getUserName() == null && !transaction.getEvents().isEmpty()) {
+            LOGGER.warn("Got transaction with null username {}", transaction);
+        }
+        else if (getConfig().getLogMiningUsernameExcludes().contains(transaction.getUserName())) {
+            LOGGER.debug("Skipping transaction with excluded username {}", transaction);
+            return;
+        }
```

Comment:
I'm concerned about the explicit `return` during the skip operation because we are explicitly skipping any changes to the metrics and offsets which I think we should do even if the transaction is skipped.  Exclusion should really only mean we don't emit events but otherwise, I think we should perform the other actions that are in this method.  Additionally, the Infinispan `handleCommit` method should also be adjusted too.

Reply:
Thanks @Naros . Right, we should not skip metrics. Updated code to fix that, and updated Infinispan class. Can you review again?